cdk/resources/containers/coap/Dockerfile

```diff
@@ -0,0 +1,66 @@
+# Define custom function directory
+ARG FUNCTION_DIR="/function"
+ARG COAP_SIMULATOR_DOWNLOAD_URL
+ARG COAP_SIMULATOR_DOWNLOAD_SECRET
+
+FROM node:20 as build-image
+
+# Include global arg in this stage of the build
+ARG FUNCTION_DIR
+ARG COAP_SIMULATOR_DOWNLOAD_URL
+ARG COAP_SIMULATOR_DOWNLOAD_SECRET
+
+# Install build dependencies
+RUN apt-get update && \
+  apt-get install -y \
+  g++ \
+  make \
+  cmake \
+  unzip \
+  libcurl4-openssl-dev \
+  wget \
+  tar
+
+# Download CoAP simulator binary
+RUN curl ${COAP_SIMULATOR_DOWNLOAD_URL} -o /tmp/coap.zip -H "Referer: ${COAP_SIMULATOR_DOWNLOAD_SECRET}" && \
```

Comment:
Not needed, but very weird to use a secret in the Referer header. Should be used in `Authorization`.

Reply:
S3 policy does not support `Authorization` header.